Landing page (#10)
PrefectHQ/prefect-openai: README.md

        <img src="https://img.shields.io/badge/discourse-browse_forum-red.svg?color=0052FF&labelColor=090422&logo=discourse" /></a>
</p>

Visit the full docs [here](https://PrefectHQ.github.io/prefect-openai) to see additional examples and the API reference.

> The prefect-dask collection makes it easy to include AI in your flows. Check out the examples below to get started!

Reviewer: ```suggestion
The prefect-openai collection makes it easy to leverage the capabilities of AI in your flows. Check out the examples below to get started!
```

Author: Haha silly me, I copied prefect-dask landing page and didn't update the name!

## Summarize tracebacks with GPT3

Tracebacks--it's quintessential in programming. They are a record of every line of code leading to the error, to help us, humans, determine what's wrong with the program and find a solution.

However, tracebacks can be extraordinarily complex, making them tedious and difficult to troubleshoot, especially for someone new to the codebase.

To streamline this process, we could add AI to the mix, to help parse through the traceback and offer a more human-readable summary of the issue.

That way, the developer can more quickly understand what went wrong and implement a fix.

> After installing `prefect-openai`, you can easily incorporate OpenAI within your flows to help you achieve the aforementioned!

Reviewer: ```suggestion
After installing `prefect-openai`, you can easily incorporate OpenAI within your flows to help you achieve the aforementioned benefits!
```

    )
    prompt = f"Summarize cause of error from this traceback: ```{traceback}```"
    summary = completion_model.submit_prompt(traceback).choices[0]["text"]
    logger.info(f"Summary of the traceback: {summary}")
    return summary

if __name__ == "__main__":
    traceback = """
        ParameterBindError: Error binding parameters for function 'summarize_traceback': missing a required argument: 'traceback'.
        Function 'summarize_traceback' has signature 'traceback: str) -> str' but received args: () and kwargs: {}.
    """
    summarize_traceback(traceback)
```

This outputs:
```bash
12:29:30.297 | INFO    | prefect.engine - Created flow run 'analytic-starling' for flow 'summarize-traceback'
12:29:32.085 | INFO    | Flow run 'analytic-starling' - Finished text completion using the 'text-curie-001' model with 113 tokens, creating 1 choice(s).
12:29:32.089 | INFO    | Flow run 'analytic-starling' - Summary of the traceback:     
This error is caused by the missing argument traceback. The function expects a traceback object as its first argument, but received nothing.
12:29:32.302 | INFO    | Flow run 'analytic-starling' - Finished in state Completed()
```

> Notice how the original traceback was kind of long, and a tad confusing, while the Curie GPT3 model summarized the issue eloquently: `This error is caused by the missing argument traceback. The function expects a traceback object as its first argument, but received nothing.`

Reviewer: ```suggestion
Notice how the original traceback was quite long and confusing, while the Curie GPT3 model summarized the issue eloquently: `This error is caused by the missing argument traceback. The function expects a traceback object as its first argument but receives nothing.`
```
